app/Services/Training/WaitingListSelfEnrolment.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Services\Training;
+
+use App\Models\Mship\Account;
+use App\Models\Mship\Qualification;
+use App\Models\Mship\State;
+use App\Models\Training\WaitingList;
+use Illuminate\Database\Eloquent\Collection;
+
+class WaitingListSelfEnrolment
+{
+    public static function canAccountEnrolOnList(
+        Account $account,
+        WaitingList $waitingList
+    ): bool {
+        if ($waitingList->includesAccount($account)) {
+            return false;
+        }
+
+        if (! $waitingList->self_enrolment_enabled) {
+            return false;
+        }
+
+        $requiresRosterMembership = $waitingList->requires_roster_membership;
+        $accountIsNotOnRoster = ! $account->onRoster();
+        if ($requiresRosterMembership && $accountIsNotOnRoster) {
+            return false;
+        }
+
+        $onlyAcceptsHomeMembers = $waitingList->home_members_only;
+        $accountIsNotHomeMember = ! $account->hasState(State::findByCode('DIVISION'));
+        if ($onlyAcceptsHomeMembers && $accountIsNotHomeMember) {
+            return false;
+        }
+
+        if ($waitingList->self_enrolment_maximum_qualification_id) {
+            $requiredQualification = Qualification::find(
+                $waitingList->self_enrolment_maximum_qualification_id
+            );
+
+            if (! $account->hasQualification($requiredQualification)) {
+                return false;
+            }
+        }
```

Comment:
```suggestion
        if ($waitingList->self_enrolment_maximum_qualification_id && ! $account->hasQualification($waitingList->self_enrolment_maximum_qualification_id)) {
                return false;
        }
```

I'll let you decide if this is better or easier to tidy up some other way. Not a huge fan of multi-level nested if statements.

Reply:
Mainly there because the min or max qualification is nullable. I'll have a think